### PR TITLE
Add explanation for APF debug output

### PR DESCRIPTION
**Pull request by mengqiy**
- Update the sample output from APF debug endpoints.
- Add clarification for columns that don't have obvious meaning.

### CODE REVIEW

**Comment by mengqiy** [comment]
cc: @wojtek-t @MikeSpreitzer @yue9944882 @tkashem to ensure accuracy.

**Comment by mengqiy** [comment]
The change can be previewed at https://deploy-preview-42075--kubernetes-io-main-staging.netlify.app/docs/concepts/cluster-administration/flow-control/#debug-endpoints

File: content/en/docs/concepts/cluster-administration/flow-control.md
Comment (windsonsea) [inline content/en/docs/concepts/cluster-administration/flow-control.md:774]:
```suggestion
  - `AdditionalLatency`: The extra time taken during the final stage of request execution.
    FinalSeats will be occupied during this time period. It does not mean any latency that a user will observe.
```

File: content/en/docs/concepts/cluster-administration/flow-control.md
Comment (windsonsea) [inline content/en/docs/concepts/cluster-administration/flow-control.md:773]:
```suggestion
  - `FinalSeats`: The number of seats will be occupied during the final stage of request execution,
    accounting for the associated WATCH notifications.
```

**Review by windsonsea — COMMENTED** [review]
some text suggestions

File: content/en/docs/concepts/cluster-administration/flow-control.md
Comment (MikeSpreitzer) [reply content/en/docs/concepts/cluster-administration/flow-control.md:774]: This looks like a good improvement to me.

**Comment by MikeSpreitzer** [comment]
Thanks for this, it is an improvement!

It would be even better if it reflected the changes merged in #119009 .  If you do not want to do that in this PR, then I can take care of it after this PR merges.

**Comment by mengqiy** [comment]
Sure! I saw it got merged. I will update my PR to reflect that.

File: content/en/docs/concepts/cluster-administration/flow-control.md
Comment (mengqiy) [reply content/en/docs/concepts/cluster-administration/flow-control.md]: Done

File: content/en/docs/concepts/cluster-administration/flow-control.md
Comment (mengqiy) [reply content/en/docs/concepts/cluster-administration/flow-control.md]: Done

**Comment by mengqiy** [comment]
@MikeSpreitzer @windsonsea PTAL
I have changed the base branch to dev-1.28 given it now reflects some changes in https://github.com/kubernetes/kubernetes/pull/119009

**Comment by MikeSpreitzer** [comment]
This looks better.
Didn't https://github.com/kubernetes/kubernetes/pull/119009 also change the row for `exempt` in dump_priority_levels ?

**Comment by mengqiy** [comment]
>Didn't https://github.com/kubernetes/kubernetes/pull/119009 also change the row for exempt in dump_priority_levels ?

No, #119009 didn't change anything regarding exempt in dump_priority_levels. But it seems the change was made in https://github.com/kubernetes/kubernetes/pull/118955

@MikeSpreitzer PTAL

**Review by MikeSpreitzer — APPROVED** [review]
/lgtm
Thanks!

**Comment by mengqiy** [comment]
@windsonsea Can you please approve or assign the right person to approve? We need to get this merged before the 1.28 release date which is EoW.

**Comment by windsonsea** [comment]
cc @sftim @tengqm

File: content/en/docs/concepts/cluster-administration/flow-control.md
Comment (tengqm) [inline content/en/docs/concepts/cluster-administration/flow-control.md:732]: Who cares about the meaning of `R`?

Comment (mengqiy) [reply content/en/docs/concepts/cluster-administration/flow-control.md:732]: K8s users should not care.
But the column name has `R`. Saying `The progress meter reading` may mislead the reader think `R` means `Reading`.
I can't find anything better. @tengqm Any suggestions?

Comment (tengqm) [reply content/en/docs/concepts/cluster-administration/flow-control.md:732]: Right. If K8s users **mostly** won't care about this debug API, maybe we should consider moving this content to the references section. It is not some basic concepts users should know anyway.

Comment (mengqiy) [reply content/en/docs/concepts/cluster-administration/flow-control.md:732]: Most k8s users should not care about this, but some cluster operators may care. When they are reading the `debug endpoint` section, it means they will want to understand what each column means here.
I'd suggest keep the column explanation inline with the `debug endpoint` section unless we want to move `debug endpoint` section to somewhere else.

Comment (sftim) [reply content/en/docs/concepts/cluster-administration/flow-control.md:732]: There is existing documentation around cluster troubleshooting.

We might want to shift that explanation to be within https://kubernetes.io/docs/tasks/debug/debug-cluster/, and change the APF docs to link to one section of those troubleshooting docs.

**Comment by mengqiy** [comment]
/hold
We may need to change the base branch.

**Comment by sftim** [comment]
> We may need to change the base branch.

Why?

**Comment by mengqiy** [comment]
From my current understanding, if this can't be merged before the release date (https://www.kubernetes.dev/resources/release/#summary). We will need to change the base branch to master since 1.28 doc will be served from master branch.

File: content/en/docs/concepts/cluster-administration/flow-control.md
Comment (mengqiy) [reply content/en/docs/concepts/cluster-administration/flow-control.md:732]: If we want to move the troubleshooting related content, we can create a new doc for APF under https://kubernetes.io/docs/tasks/debug/debug-cluster/ and we will move https://v1-23.docs.kubernetes.io/docs/concepts/cluster-administration/flow-control/#diagnostics and https://v1-23.docs.kubernetes.io/docs/concepts/cluster-administration/flow-control/#debug-endpoints there while we keep metrics section as is.
If this make sense, I can make the change.

**Comment by sftim** [comment]
If this doesn't land before the v1.28 release freeze, let's change the base branch to _master_

**Comment by mengqiy** [comment]
@sftim @tengqm PTAL

**Comment by tengqm** [comment]
I don't think this diagnostics page is suitable to be a task. There are no steps to follow.
It is, in my mind, a reference for a small group of users who really understand not what APF does, but also how it is implemented. It is something only some advanced hackers need to know.
By the way, the page content looks pretty good.

File: content/en/docs/tasks/debug/debug-cluster/flow-control.md
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/flow-control.md:1]: I agree with https://github.com/kubernetes/website/pull/42075#issuecomment-1694676318

@mengqiy if you rewrite this to match other pages in the same section in terms of describing steps someone should take, it'll fit better.

Please also set a `weight` for the new page (I would pick a new number higher than any existing weight in the same section).

File: content/en/docs/tasks/debug/debug-cluster/flow-control.md
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/flow-control.md:16]: I imagine you also need API Priority and Fairness enabled?

File: content/en/docs/tasks/debug/debug-cluster/flow-control.md
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/flow-control.md:10]:
```suggestion
[API Priority and Fairness](/docs/concepts/cluster-administration/flow-control/)
documentation.
```

File: content/en/docs/tasks/debug/debug-cluster/flow-control.md
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/flow-control.md:48]: What format is that in? Is it CSV, for example (and defined as such)? If it is, let's say so.

File: content/en/docs/tasks/debug/debug-cluster/flow-control.md
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/flow-control.md:42]: What access do you need in order to fetch that debug info?

File: content/en/docs/tasks/debug/debug-cluster/flow-control.md
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/flow-control.md:148]:
```suggestion
At `-v=3` or more verbosity, the API server outputs an httplog line for every
```

File: content/en/docs/tasks/debug/debug-cluster/flow-control.md
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/flow-control.md:160]: Where is that log written / how can a cluster operator view it?

File: content/en/docs/tasks/debug/debug-cluster/flow-control.md
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/flow-control.md:172]: I would mention whether these appear in audit logs. AIUI they don't, and you can't enable that - have I got this right?

File: content/en/docs/tasks/debug/debug-cluster/flow-control.md
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/flow-control.md:178]: Skip these details here.

File: content/en/docs/tasks/debug/debug-cluster/flow-control.md
Comment (sftim) [reply content/en/docs/tasks/debug/debug-cluster/flow-control.md:16]:
```suggestion
- You must have the `APIPriorityAndFairness`
  [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
  enabled in your cluster (this feature is enabled by default for Kubernetes versions
  1.20 through to {{< skew currentVersion >}}, but you can explicitly disable it).

```

File: content/en/docs/tasks/debug/debug-cluster/flow-control.md
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/flow-control.md:26]: _optional tweak_
```suggestion
names are not included in these headers (to avoid revealing details in case the
requesting user does not have permission to view them). When debugging, you
can use a command such as:

File: content/en/docs/tasks/debug/debug-cluster/flow-control.md
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/flow-control.md:34]: Also consider:
```shell
kubectl get flowschemas -o jsonpath='{range .items[?(.metadata.uid=262bab1a-1c79-11ea-8e23-42010a800016)]}{.metadata.name}{end}'
```
or similar

**Comment by reylejano** [comment]
@mengqiy , please rebase

**Comment by mengqiy** [comment]
Thanks for the feedback!
I was on vacation in the past weeks. I will address the comments and rebase.

File: content/en/docs/tasks/debug/debug-cluster/flow-control.md
Comment (mengqiy) [reply content/en/docs/tasks/debug/debug-cluster/flow-control.md:1]: There are not any steps here. So I have moved it the reference doc as @tengqm suggested.
But I'm not sure what weight I should set. I don't quite understand how it works.

File: content/en/docs/tasks/debug/debug-cluster/flow-control.md
Comment (mengqiy) [reply content/en/docs/tasks/debug/debug-cluster/flow-control.md]: No longer need this part since it's not a troubleshooting doc any more.

File: content/en/docs/tasks/debug/debug-cluster/flow-control.md
Comment (mengqiy) [reply content/en/docs/tasks/debug/debug-cluster/flow-control.md]: Done

File: content/en/docs/tasks/debug/debug-cluster/flow-control.md
Comment (mengqiy) [reply content/en/docs/tasks/debug/debug-cluster/flow-control.md]: Updated

File: content/en/docs/tasks/debug/debug-cluster/flow-control.md
Comment (mengqiy) [reply content/en/docs/tasks/debug/debug-cluster/flow-control.md]: In the APIServer log.
I have added it in the doc

File: content/en/docs/tasks/debug/debug-cluster/flow-control.md
Comment (mengqiy) [reply content/en/docs/tasks/debug/debug-cluster/flow-control.md]: Updated.
Will not appear in audit log.

File: content/en/docs/tasks/debug/debug-cluster/flow-control.md
Comment (mengqiy) [reply content/en/docs/tasks/debug/debug-cluster/flow-control.md]: I'm not a jsonpath expert. But this doesn't seems to be right to output a list of <name, uid> pairs.

**Comment by mengqiy** [comment]
@sftim @tengqm Rebased and updated. PTAL

File: content/en/docs/reference/flow-control/_index.md
Comment (tengqm) [inline content/en/docs/reference/flow-control/_index.md:4]: Since this is a "no_list" subdirectory, maybe we can create a `reference/debug-cluster` directory and put this page under that directory. When we name this file as `reference/debug-cluster/flow-control.md`, we are making room for future topics related to debugging cluster.

File: content/en/docs/reference/flow-control/_index.md
Comment (tengqm) [inline content/en/docs/reference/flow-control/_index.md:36]:
```suggestion
With the `APIPriorityAndFairness` feature enabled, the `kube-apiserver`
```

File: content/en/docs/reference/flow-control/_index.md
Comment (tengqm) [inline content/en/docs/reference/flow-control/_index.md:41]:
```suggestion
Permissions can be granted if needed following the [RBAC](/docs/reference/access-authn-authz/rbac/)
```

File: content/en/docs/reference/flow-control/_index.md
Comment (tengqm) [inline content/en/docs/reference/flow-control/_index.md:151]: httplog is a new term to me ...

File: content/en/docs/reference/flow-control/_index.md
Comment (tengqm) [inline content/en/docs/reference/flow-control/_index.md:152]:
```suggestion
request in the API server log, and it includes the following attributes.
```

File: content/en/docs/reference/flow-control/_index.md
Comment (tengqm) [inline content/en/docs/reference/flow-control/_index.md:159]:
```suggestion
  execution (accounting for the associated `watch` notifications) of the
```

File: content/en/docs/reference/flow-control/_index.md
Comment (tengqm) [inline content/en/docs/reference/flow-control/_index.md:171]:
```suggestion
For client using `klog`, use verbosity `-v=8` or higher to view these headers.
```

File: content/en/docs/reference/flow-control/_index.md
Comment (mengqiy) [reply content/en/docs/reference/flow-control/_index.md]: It is provided by https://github.com/kubernetes/apiserver/blob/master/pkg/server/httplog/doc.go. 
It allows you to add key-value pairs which is heavily used here. https://github.com/kubernetes/apiserver/blob/83bd32f90ba0c7cb082e2605a21e1a9f183481cd/pkg/server/httplog/httplog.go#L227

File: content/en/docs/reference/flow-control/_index.md
Comment (mengqiy) [reply content/en/docs/reference/flow-control/_index.md]: Moved to `reference/debug-cluster`. Please let me know if `weight` and other configuration make sense.

**Comment by mengqiy** [comment]
@tengqm @sftim PTAL

**Comment by mengqiy** [comment]
Need someone to apply the LGTM label.
The content has been LGTM'd [before](https://github.com/kubernetes/website/pull/42075#pullrequestreview-1566263208)

**Comment by windsonsea** [comment]
Thanks.
/lgtm

**Pull request merged**
